### PR TITLE
ga4 tracking for filter/sorting UI interaction

### PR DESCRIPTION
**Pull request by alexbowen**
For new search page implementation, this PR adds ga4 initialisation and attributes to filters and sorting UI to meet requirements listed here:

https://docs.google.com/spreadsheets/d/1dRoxYPatZiNJGYZEc6E_FZ5xeD_aWQdGxQfYK1dUFq0

- Filter panel is a simple custom accordion and tracking of opening and closing is implemented using the `data-ga4-expandable` attribute which adds opened/closed to the action property of the event data.
- Filter sections are standard HTML details elements and tracking of opening and closing is implemented again using the `data-ga4-expandable` attribute which adds opened/closed to the action property of the event data.
- Interaction with form elements that apply filtering/sorting adds ga4 events using the existing `finder-frontend/app/assets/javascripts/analytics-ga4/ga4-finder-tracker.js` script. The new `all-content-finder` search page uses a form with id to identify it and initialise the tracking scripts if the form is present (and cookie consent has been granted from the user).
- Filter/sorting remove links and clear all links (`filter-summary` component) use `ga4-event-tracker` attribute and standard child attributes to track when they are clicked.

_No visual changes._

### CODE REVIEW

**Comment by andysellick** [comment]
@alexbowen it's possible that some existing functionality using the `data-ga4-expandable` attribute might help you here, see: https://github.com/alphagov/govuk_publishing_components/blob/main/docs/analytics-ga4/trackers/ga4-event-tracker.md#advanced-use

**Comment by alexbowen** [comment]
> @alexbowen it's possible that some existing functionality using the `data-ga4-expandable` attribute might help you here, see: https://github.com/alphagov/govuk_publishing_components/blob/main/docs/analytics-ga4/trackers/ga4-event-tracker.md#advanced-use

Yeah i was aware of this. Will have a look again at this now i am more familiar with ga4 stuff. was have trouble getting `data-ga4-expandable` to work before but am about to revisit

File: app/views/finders/all_content_finder_facets/_taxon_facet.html.erb [hunk 1]
```diff
@@ -4,22 +4,22 @@
   id: "facet_taxon_#{taxon_facet.key}",
   index_section: index,
   index_section_count: count,
+  data_attributes: { "ga4-index": { index_section: index + 1, index_section_count: count } },
+  change_category: "update-filter select",
 } do %>
-  <div class="js-all-content-finder-taxonomy-select" data-ga4-change-category="update-filter select" data-ga4-section="Topic">
```
[the remark this answers is not in the document]
Comment (alexbowen) [reply app/views/finders/all_content_finder_facets/_taxon_facet.html.erb:8]: there was a reason i did this at some point and i did indeed mean to put that class the the container but must have got distracted! so the class is on the component now as that div seems uneccessary

File: spec/components/filter_section_spec.rb [hunk 2]
```diff
@@ -56,4 +56,22 @@ def render_component(locals)
...
+    button_event_attributes = {
+      event_name: "select_content",
+      type: "finder",
+      section: heading_text,
+      text: heading_text,
+      index_section: 1,
+      index_section_count: 0,
+    }
+
+    render_component(heading_text:, data: button_event_attributes.to_json)
```
[the remark this answers is not in the document]
Comment (alexbowen) [reply spec/components/filter_section_spec.rb:72]: similar i think a hangover from something early in implementing this, it is included all over tests and have removed all as yes they do absolutely nothing

File: app/assets/javascripts/application.js [hunk 3]
```diff
@@ -57,3 +57,20 @@ if ($allContentFinderTaxonomySelect) {
...
+      ga4ChangeCategory = ga4ChangeCategory.getAttribute('data-ga4-change-category')
+
+      const consentCookie = GOVUK.getConsentCookie()
+
+      if (consentCookie && consentCookie.usage) {
+        GOVUK.analyticsGa4.Ga4FinderTracker.trackChangeEvent(event.target, ga4ChangeCategory)
+      }
+    }
+  })
+}
```
[the remark this answers is not in the document]
Comment (alexbowen) [reply app/assets/javascripts/application.js:76]: Yes i was thinking to put this in a more isolated place than here but was in 2 minds about a few places where exactly. so I have left as is for the purposes of this PR, but agreed its a bit lazy just whacking it here in the long run

**Review by csutter — APPROVED** [review]

**Pull request merged**